backend/lib/endpoints/posts.js

```diff
@@ -37,6 +40,12 @@ async function routes(app) {
           localField: "_id",
         },
       },
+      {
+        $skip: skip || 0,
+      },
+      {
+        $limit: limit || DEFAULT_LIMIT,
+      },
```

Comment:
That's better!

I would just recommend putting these two steps before the lookup stage there.
So the pipeline would look something like this:
```
    aggregates.push(
      {
        $skip: skip || 0,
      },
      {
        $limit: limit || DEFAULT_LIMIT,
      },
      {
          $lookup: {
            as: "comments",
            foreignField: "postId",
            from: "comments",
            localField: "_id",
         },
       },
       // rest of the pipeline as is
    )
```
It's just changing the execution order.
This way the lookup will only be executed for the posts that are within our page range.